### PR TITLE
User may want to skip tidy check sometimes

### LINKED ISSUE #113135: suggest `--no-verify` if the git hook fails
https://github.com/rust-lang/rust/blob/f6d58eaad3d2b426862a65c5dec9391544d66948/src/etc/pre-push.sh#L16-L17

https://discord.com/channels/273534239310479360/957720175619215380/1123682277243768863

### PR DESCRIPTION
**Pull request by chenyukang**
Fixes #113135

### CODE REVIEW

**Comment by Kobzol** [comment]
This won't have an effect for people that already have the hook installed, but that's not such an issue, since this PR is aimed towards future new contributors anyway. I have tested it locally that it works.

@bors r+ rollup

File: src/etc/pre-push.sh [hunk 1]
```diff
@@ -15,3 +15,7 @@ echo "Running pre-push script $ROOT_DIR/x test tidy"
 
 cd "$ROOT_DIR"
 ./x test tidy --set build.locked-deps=true
+if [ $? -ne 0 ]; then
```
Comment (tgross35) [inline src/etc/pre-push.sh:18]:
```suggestion
res=ok
./x test tidy --set build.locked-deps=true || res=fail
if [ "$res" != 0 ]; then
```

Probably too late, but you could do something like this to not need to disable the `-e` flag. Not really needed anyway in this tiny script though

Comment (chenyukang) [reply src/etc/pre-push.sh:18]: nice trick!
but it should be:
```bash
if [ "$res" != "ok" ]; then
```
right?

Comment (tgross35) [reply src/etc/pre-push.sh:18]: Whoops, you're right!

**Pull request merged**